### PR TITLE
Support of FastDDS 2.x

### LINKED ISSUE #19: Build failing with Fast-DDS
This line causes an error

https://github.com/ros2/performance_test/blob/df75c932e4c4e7c22cb57b8f120ac04660d9e177/performance_test/src/communication_abstractions/fast_rtps_communicator.hpp#L29

Because that API was deleted in https://github.com/eProsima/Fast-DDS/pull/1227.

---

I only could reproduce locally, I'm not sure why [nightlies](http://build.ros2.org/view/Fci/job/Fci__nightly-performance_ubuntu_focal_amd64) aren't failing.
We're using [2.0.x](https://github.com/ros2/ros2/blob/1a54bc524f00e4ac59de9b665689ea1f7fdfadf5/ros2.repos#L37) Fast-DDS branch in Foxy ...

We're also using `-DPERFORMANCE_TEST_FASTRTPS_ENABLED=1` cmake option there.

### PR DESCRIPTION
**Pull request by richiware**
Fixes #19.

### CODE REVIEW

**Comment by ivanpauno** [comment]
We're using `run_on_builfarm` branch, so this PR should target it.

File: performance_test/src/communication_abstractions/resource_manager.cpp [hunk 1]
```diff
@@ -85,7 +85,7 @@ eprosima::fastrtps::Participant * ResourceManager::fastrtps_participant() const
   disc_config.m_simpleEDP.use_PublicationWriterANDSubscriptionReader = true;
   disc_config.leaseDuration = eprosima::fastrtps::c_TimeInfinite;
 #endif
-  PParam.rtps.builtin.domainId = m_ec.dds_domain_id();
+  PParam.domainId = m_ec.dds_domain_id();
```
Comment (ivanpauno) [inline performance_test/src/communication_abstractions/resource_manager.cpp:88]: is this change compatible with the version of Fast-RTPS used in eloquent?
This code is used for both Foxy and eloquent.

Comment (cottsay) [reply performance_test/src/communication_abstractions/resource_manager.cpp:88]: This is a good point, I think we should try to make the change compatible with the supported releases if possible. If not, we'll have to create a separate branch for eloquent, which would be unfortunate.

Comment (richiware) [reply performance_test/src/communication_abstractions/resource_manager.cpp:91]: Fixed.

Comment (richiware) [reply performance_test/src/communication_abstractions/resource_manager.cpp:91]: Fixed

Comment (ivanpauno) [reply performance_test/src/communication_abstractions/resource_manager.cpp:91]: There's no compile option being set in the CMakeLists.txt called `FASTRTPS_VERSION_MAJOR`, e.g. https://github.com/ros2/performance_test/blob/ad8a90a6333ff6008fd195c3cb36ccc90e5eb2dd/performance_test/CMakeLists.txt#L61

We need some extra logic in the `CMakelists.txt` for this to work

**Comment by richiware** [comment]
That define is provided by Fast-DDS ([fastrtps/config.h](
https://github.com/eProsima/Fast-DDS/blob/9125239274040384af8b1ad981817333fc4f9d58/include/fastrtps/config.h.in#L18)).
No changes should be made.

File: performance_test/src/communication_abstractions/resource_manager.cpp (hunk 1 above)
Comment (MiguelCompany) [reply performance_test/src/communication_abstractions/resource_manager.cpp:91]: > There's no compile option being set in the CMakeLists.txt called `FASTRTPS_VERSION_MAJOR`, e.g.

This becomes available whenever you include something from Fast-DDS.
It is done the same way on [rmw_fastrtps](https://github.com/ros2/rmw_fastrtps/blob/4ade4a136c29c5885a2debea8b173ba29dc6ebcd/rmw_fastrtps_shared_cpp/src/participant.cpp#L168)

Comment (ivanpauno) [reply performance_test/src/communication_abstractions/resource_manager.cpp:91]: Yes, sorry. I saw the other reply in https://github.com/ros2/performance_test/pull/20#issuecomment-646842473.
I'm waiting for a reply [here](https://github.com/ros2/performance_test/issues/19#issuecomment-646852500) before merging this.

**Comment by ivanpauno** [comment]
Nightly jobs are currently broken because of this.
I'm not sure why they started to fail on [June 19th](http://build.ros2.org/view/Fci/job/Fci__nightly-performance_ubuntu_focal_amd64/82/) when https://github.com/eProsima/Fast-DDS/pull/1227 was merged on May 25th (maybe, we're using released versions in Fci??).

I'm going to merge this to see if it unblocks the nightly jobs.
FYI @cottsay

**Comment by ivanpauno** [comment]
Thanks for the PR @richiware !!

**Comment by ivanpauno** [comment]
Oh :man_facepalming:, I forgot this is targeting the incorrect branch: https://github.com/ros2/performance_test/pull/20#issuecomment-645351593.

I will open another PR targeting the correct one.

**Comment by cottsay** [comment]
I really think there should be a PR upstream as well. We really want to kill this fork.

**Comment by ivanpauno** [comment]
> I really think there should be a PR upstream as well. We really want to kill this fork.

Yeah, I'm planning to do that too.

**Pull request merged**